File: backend/animal-registry/src/main/java/com/animalshelter/animalregistry/dto/RagSearchResponse.java
package com.animalshelter.animalregistry.dto;

import java.util.List;

public record RagSearchResponse(
        String query,
        String answer,
        List<MatchedAnimal> matchedAnimals,
        int totalMatches
) {
    public record MatchedAnimal(
            String id,
            String name,
            String category,
            String breed,
            String status,
            String description,
            double relevanceScore
    ) {}
}

File: frontend/src/app/dto/animal/RagSearchResponse.ts
export interface RagSearchMatch {
  id: string;
  name: string;
  category: string;
  breed: string;
  status: string;
  description: string;
  relevance_score: number;
}

export interface RagSearchResponse {
  query: string;
  answer: string;
  matched_animals: RagSearchMatch[];
  total_matches: number;
}
